client/ayon_houdini/plugins/publish/increment_current_file.py

```diff
@@ -46,9 +48,19 @@ def process(self, context):
         host = registered_host()
         current_file = host.current_file()
         if context.data["currentFile"] != current_file:
-            raise KnownPublishError(
-                "Collected filename mismatches from current scene name."
+            raise PublishError(
```

Comment:
Just out of curiosity why there is a specific check on the currentFile in both increment and save file? Also why do we need to save the workfile twice for render and workfile families?

Reply:
Honestly, I don't know. 
And, not only that but a lot of  plugins have the same checks e.g. alembic validators. 
I've created this issue yesterday 
also, I've added your note. thank you.
https://github.com/ynput/ayon-houdini/issues/194

Reply:
> Just out of curiosity why there is a specific check on the currentFile in both increment and save file? Also why do we need to save the workfile twice for render and workfile families?

Where are we saving twice?

We're saving (before incrementing)... and then incrementing the workfile so the user doesn't continue working in the original version.

---

Also, they both check the same thing to ensure that if the plug-ins would be operating in isolation of each other this thing is checked, because they don't NEED to run together.

Reply:
Should we refactor the error in `IncrementCurrentFile` to `KnownPublishError`. 
as a user I can change the file name between collection (validation to be percise) and before extraction by clicking validate button. and change the file then continue publishing. 

But, during integration (`IncrementCurrentFile`), I'm not sure how can I break it.

Reply:
Also, I'm thinking about changing the comment in the code to 
```
# Filename must not have changed since extraction.
```

Reply:
Both sound good.

Reply:
https://github.com/ynput/ayon-houdini/pull/193/commits/25850ea18a7a3c3326edcadfed693b5d94a3f391